src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/EventLogException.cs

```diff
@@ -50,7 +51,11 @@ internal static void Throw(int errorCode)
         public EventLogException() { }
         public EventLogException(string message) : base(message) { }
         public EventLogException(string message, Exception innerException) : base(message, innerException) { }
-        protected EventLogException(int errorCode) { _errorCode = errorCode; HResult = errorCode; }
+        protected EventLogException(int errorCode)
+        {
+            _errorCode = errorCode;
+            HResult = Win32Marshal.MakeHRFromErrorCode(errorCode);
```

Comment:
Given Win32Marshal brings in all these strings, it might be better to simply do it yourself here. I see that in a few places in our code.

Better still, I found we have this which you can bring in instead
https://github.com/dotnet/runtime/blob/39078f54011cd78acc378007b024b5cb0f251d47/src/libraries/Common/src/Interop/Windows/Interop.HRESULT_FROM_WIN32.cs#L10